src/lib/operators/import.cpp

```diff
@@ -66,15 +70,21 @@ std::shared_ptr<const Table> Import::_on_execute() {
 
   Hyrise::get().storage_manager.add_table(_tablename, table);
 
-  // must match ImportNode::output_expressions
+  // If a table encoding is specified, encode the table accordingly.
+  if (_table_encoding) {
+    const auto chunk_encoding_spec =
+        ChunkEncodingSpec(table->column_count(), SegmentEncodingSpec(_table_encoding.value()));
+    ChunkEncoder::encode_all_chunks(Hyrise::get().storage_manager.get_table(_tablename), chunk_encoding_spec);
+  }
```

Comment:
@Bouncner what's your take on "default encoding"? As of now, `.tbl` and `.csv` files are stored `Unencoded`, `.bin`s in the encoding of the input.

While I would find it counterintuitive to re-encode an already encoded binary of no encoding is explicitly requested in the SQL statement (and the binary's encoding might be a lot more fine-grained), I think a default (e.g., `Dictionary`) for `.csv` and `.tbl` makes sense.

Reply:
I was not aware we do not encudre CSV files to be honest. I would definitely favor our default encoding (dict) for imported files.